src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int capacity = DEFAULT_CAPACITY;
+    private Node<K, V>[] table = new Node[capacity];
+    private int size = 0;
+    private int threshold = 0;
 
     @Override
     public void put(K key, V value) {
+        int hash = key == null ? 0 : Math.abs(key.hashCode());
+        int index = Math.abs(hash) % capacity;
+        if (index < 0) {
+            index += capacity;
```

Comment:
The index can't be negative because you're using `Math.abs(hash)`. So you don't need to check if the index is less than 0 and add capacity to it.